src/create-client.ts

```diff
@@ -334,8 +334,9 @@ An authorization_code was supplied for a login which did not originate at the ap
         beginningState.tag !== "INITIAL" &&
           this.#onRefreshFailure &&
           this.#onRefreshFailure({ signIn: this.signIn.bind(this) });
+
+        this.#state = { tag: "ERROR" };
```

Comment:
the downside to the #private methods/instance variables we're using these days is that they aren't straightforward to mock. i want to write a test for this but i don't think there is a decent way to do this. (i thought maybe jest.mock the new http client, but that applies to the whole test file so it's not ideal.

Reply:
True, though if you are trying to mock things about the HTTP client (like how it behaves with errors), it'd be preferable to use [these nock style mocks](https://github.com/workos/authkit-js/blob/22adb78bc85e6058c97979ad679c308262edb28a/src/create-client.test.ts#L142-L147) instead.